client/ayon_core/pipeline/context_tools.py

```diff
@@ -617,7 +617,18 @@ def version_up_current_workfile():
     last_workfile_path = get_last_workfile(
         work_root, file_template, data, extensions, True
     )
-    new_workfile_path = version_up(last_workfile_path)
+    # `get_last_workfile` will return the first expected file version
+    # if no files exist yet. In that case, if they do not exist we will
+    # want to save v001
+    new_workfile_path = last_workfile_path
     if os.path.exists(new_workfile_path):
         new_workfile_path = version_up(new_workfile_path)
+
+    # Raise an error if the parent folder doesn't exist as `host.save_workfile`
+    # is not supposed/able to create missing folders.
+    parent_folder = os.path.dirname(new_workfile_path)
+    if not os.path.exists(parent_folder):
+        raise AssertionError(
+            f"Folder {parent_folder} does not exist yet.")
+
```

Comment:
If I get it correctly, this is what we need (requires to implement `MissingWorkfile`).
```suggestion
    if not os.path.exists(last_workfile_path):
        raise MissingWorkfile(
            f"No workfile found for current context.")

    new_workfile_path = version_up(last_workfile_path)
```

Reply:
This will generate many issues I'd say - because we have quite a few places in the code that expect this to return the first valid version number, if none exists yet. 

E.g. all the "Version Up Workfile" file menu entries are allowed to save the initial version (or the new version) if the current scene name is unsaved and there's no workfile yet.

Reply:
Ok, so based on https://github.com/ynput/ayon-core/pull/1141#issuecomment-2678368973 , this suggestion is actually better?

1. Check if last workfile is existing file, if not, then there is nothing to update?
2. Continue as before...

Reply:
Or if the workdir existence is the issue, then please first check the parent folder, before running version up.
```python
    if not os.path.exists(os.path.dirname(last_workfile_path)):
        raise MissingWorkdir(
            "Workdir for current context does not exist.")

    new_workfile_path = last_workfile_path
    if os.path.exists(new_workfile_path):
        new_workfile_path = version_up(new_workfile_path)
```
Again, technically the same, but more readable and understandable.

Reply:
And if the issue is missing workdir, shouldn't we just create it instead of raising error?

Reply:
We could, but creating the folder isn't sufficient. Because technically we should then "initialize the work area" including creating the "Work folders" as per settings, etc. to ensure the studio's work area is actually initialized as intended?

Reply:
What would be the consequences of just creating the folder and saving the workfile?
will that cause skipping initializing the work directory later?

Reply:
> What would be the consequences of just creating the folder and saving the workfile?
> will that cause skipping initializing the work directory later?

It may mean that we can't rely anymore on the work directory being correctly initialized on that particular run at least, even if it were created later. As such, we somehow are introducing a way where the scene file _may_ be generated and saved there without the work directory being initialized in the way a studio might have configured it should. Resulting in confusion.

So either, we disallow this and error here. Or we enforce this to initialize fully the work directory.

Reply:
Let just disallow it. _whispering, It's currently disallowed._